test_apps/smokeTests/gridTools.cpp

```diff
@@ -0,0 +1,436 @@
+    }
+
+    float *buf = new float[nblocks * block_size];
+
+    Heap.push_back(buf);
+
+    std::vector <float *> blks;
+    for (size_t i=0; i<nblocks; i++) {
+        blks.push_back(buf + i*block_size);
+    }
+    return(blks);
+}
+
+void MakeTriangle( Grid* grid, float minVal, float maxVal ) {
+    std::vector< size_t > dims = grid->GetDimensions();
+    size_t x = dims[X];
+    size_t y = dims[Y];
+    size_t z = 1;
+    if ( dims.size() == 3 ) 
+        z = dims[Z];
+    
+    float value = minVal;
+    for ( size_t k=0; k<z; k++ ) {
+        for ( size_t j=0; j<y; j++ ) {
+            for ( size_t i=0; i<x; i++ ) {
+                value = value==minVal ? maxVal : minVal;
+                grid->SetValueIJK( i, j, k, value );
+            }
+        }
+    }
+}
+
+void MakeConstantField( Grid* grid, float value ) {
+    std::vector< size_t > dims = grid->GetDimensions();
+    size_t x = dims[X];
+    size_t y = dims[Y];
+    size_t z = 1;
+    if ( dims.size() == 3 ) 
+        z = dims[Z];
+
+    for ( size_t k=0; k<z; k++ ) {
+        for ( size_t j=0; j<y; j++ ) {
+            for ( size_t i=0; i<x; i++ ) {
+                grid->SetValueIJK( i, j, k, value );
+            }
+        }
+    }
+}
+
+void MakeRamp( Grid* grid, float minVal, float maxVal ) {
+    std::vector< size_t > dims = grid->GetDimensions();
+    size_t x = dims[X];
+    size_t y = dims[Y];
+    size_t z = 1;
+    if ( dims.size() == 3 ) 
+        z = dims[Z];
+
+    float increment = (maxVal - minVal) / ((x-1)*(y-1)*(z-1));
+
+    float value = minVal;
+    for ( size_t k=0; k<z; k++ ) {
+        for ( size_t j=0; j<y; j++ ) {
+            for ( size_t i=0; i<x; i++ ) {
+                grid->SetValueIJK( i, j, k, value );
+                value += increment;
+            }
+        }
+    }
+}
+
+void MakeRampOnAxis( Grid* grid, float minVal, float maxVal, size_t axis=X ) {
+    std::vector< size_t > dims = grid->GetDimensions();
+    size_t x = dims[X];
+    size_t y = dims[Y];
+    size_t z = 1;
+    if ( dims.size() == 3 ) 
+        z = dims[Z];
+
+    float xIncrement = axis==X ? (maxVal - minVal) / (dims[X]-1) : 0;
+    float yIncrement = axis==Y ? (maxVal - minVal) / (dims[Y]-1) : 0;
+    float zIncrement = axis==Z ? (maxVal - minVal) / (dims[Z]-1) : 0;
+
+    float value = minVal;
+    for ( size_t k=0; k<z; k++ ) {
+        value = axis==Y ? minVal : value;      // reset value if we're ramping on Y
+        for ( size_t j=0; j<y; j++ ) {
+            value = axis==X ? minVal : value;  // reset value if we're ramping on X
+            for ( size_t i=0; i<x; i++ ) {
+                grid->SetValueIJK( i, j, k, value );
+                value += xIncrement;
+            }
+            value += yIncrement;
+        }
+        value += zIncrement;
+    }
+}
+
+// This function iterates across all nodes in a grid; making comparisons between the
+// data values returned by the functions GetValueAtIndex() and GetValue().
+int CompareIndexToCoords( 
```

Comment:
The convention we use for error handling throughout VAPOR is that functions that report errors via MyBase return integer status (or NULL in the case of pointers), and functions that do not report errors return a boolean to indicate success or failure.

Reply:
Fixed